### ISSUE TITLE
Warn when iteratee method uses too many arguments

### ISSUE THREAD

**Issue by jfmengels** [jfmengels/eslint-plugin-lodash-fp#28]
Rule name to be found.

In Lodash/fp, many methods with an iteratee have their iteratee called with a fixed arity, often one less than what they're used to in vanilla Lodash. It would be nice to warn the user of this.

Example:
```js
// Vanilla
_.map([1, 2, 3], function(value, index) {
  return value + index; // OK, index will be 0, 1, 2 in order
});

// FP
_.map(function(value, index) {
  return value + index; // Not OK, as index is never passed (always `undefined`)
}, [1, 2, 3]);
```
